### PR TITLE
Top level function for migrations

### PR DESCRIPTION
**Pull request by marius-serban**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have rebased on `develop` squashed my commits into 1 commit.
- [x] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/HowToContribute).

## What's in this pull request?
Added a top level function that performs progressive migrations.
The function is a bit large, it could be broken down but not sure what's your preference @jessesquires .

### CODE REVIEW

File: Source/CoreDataExtensions.swift [hunk 1]
```diff
@@ -199,3 +199,83 @@ public func resetStack(stack: CoreDataStack,
...
+ */
+public func migrate(model: CoreDataModel) throws {
+
+    func findModelsInBundle(bundle: NSBundle) -> [NSManagedObjectModel] {
+        guard let modelDirectoryURLs = bundle.URLsForResourcesWithExtension("momd", subdirectory: nil) else {
+            return []
+        }
+
+        return Array(modelDirectoryURLs.flatMap { $0.lastPathComponent }.flatMap { bundle.URLsForResourcesWithExtension("mom", subdirectory: $0) }.flatten())
+            .flatMap { NSManagedObjectModel(contentsOfURL: $0) }
```
Comment (jessesquires) [inline Source/CoreDataExtensions.swift]: let's breakup this return into multiple steps with descriptive variable names so its easier to read :smile:

**Comment by jessesquires** [comment]
Thanks @marius-serban ! :smile: :tada: 

At first glance, this looks great! Before I review in-depth, let's do some re-organization:

- Move new code from `CoreDataExtensions.swift` to a new file, `Migrate.swift`
- Pull the inner functions out, try to breakup the remaining parts of the function into smaller ones
- All helper functions should be `internal` and tested if possible/if it makes sense, otherwise let's make them `private`

**Comment by jessesquires** [comment]
Thanks @marius-serban ! :smile: :tada: 

At first glance, this looks great! Before I review in-depth, let's do some re-organization:

- Move new code from `CoreDataExtensions.swift` to a new file, `Migrate.swift`
- Pull the inner functions out, try to breakup the remaining parts of the function into smaller ones
- All helper functions should be `internal` and tested if possible/if it makes sense, otherwise let's make them `private`

**Comment by jessesquires** [comment]
@marius-serban - Is this ready to go? :smile:

**Comment by marius-serban** [comment]
yup! made those changes

--
Marius

**Comment by jessesquires** [comment]
:tada: Thanks so much @marius-serban !

I'll be merging soon. here's the plan:

1. Release 2.2.1
3. Merge this and release 3.0 with Swift 2.2

**Comment by marius-serban** [comment]
+1

**Comment by jessesquires** [comment]
@marius-serban - just hand another look at this after merging. And I'm like:



Please keep contributing! :smile:

**Comment by jessesquires** [comment]
hey @marius-serban ! 

i'm going through PRs to add contributors as collaborators on this project. just sent you an invite! 😄

**Pull request merged**